app/bundles/LeadBundle/Resources/views/Field/list.html.twig

```diff
@@ -23,122 +23,174 @@
-                          {% endif %}
-                          {% if item.isFixed %}
-                              <i class="ri-lock-fill" data-toggle="tooltip" data-placement="left" title="{{ 'mautic.lead.field.tooltip.fixed'|trans }}"></i>
-                          {% endif %}
-                          {% if item.isListable %}
-                              <i class="ri-list-check " data-toggle="tooltip" data-placement="left" title="{{ 'mautic.lead.field.tooltip.listable'|trans }}"></i>
-                          {% endif %}
-                          {% if item.isPubliclyUpdatable %}
-                              <i class="ri-earth-line text-danger " data-toggle="tooltip" data-placement="left" title="{{ 'mautic.lead.field.tooltip.public'|trans }}"></i>
-                          {% endif %}
-                          {% if item.isUniqueIdentifer %}
-                              <i class="ri-key-2-line " data-toggle="tooltip" data-placement="left" title="{{ 'mautic.lead.field.tooltip.isuniqueidentifer'|trans }}"></i>
-                          {% endif %}
-                          {% if item.isIsIndex %}
-                              <i class="ri-key-2-line " data-toggle="tooltip" data-placement="left" title="{{ 'mautic.lead.field.tooltip.isindex'|trans }}"></i>
-                          {% endif %}
-                      </td>
-                  </tr>
-              {% endfor %}
-              </tbody>
-          </table>
-      </div>
-      <div class="panel-footer">
-          {{ include('@MauticCore/Helper/pagination.html.twig', {
-                  'totalItems': totalItems,
-                  'page': page,
-                  'limit': limit,
-                  'baseUrl': path('mautic_contactfield_index'),
-                  'sessionVar': 'leadfield',
+  {# Group items by object #}
+  {% set groupedItems = {} %}
+  {% set objectTotals = {} %}
+  {% set objects = ['lead', 'company'] %}
+
+  {% for item in items %}
+      {% set objectKey = item.object %}
+      {% if objectKey not in objects %}
+          {% set objects = objects|merge([objectKey]) %}
+      {% endif %}
+      {% if groupedItems[objectKey] is not defined %}
+          {% set groupedItems = groupedItems|merge({(objectKey): []}) %}
+          {% set objectTotals = objectTotals|merge({(objectKey): 0}) %}
+      {% endif %}
+      {% set groupedItems = groupedItems|merge({
+          (objectKey): groupedItems[objectKey]|merge([item])
+      }) %}
+      {% set objectTotals = objectTotals|merge({
+          (objectKey): objectTotals[objectKey] + 1
+      }) %}
+  {% endfor %}
+
+  {# Prepare tabs array #}
+  {% set tabsContent = [] %}
+  {% for objectKey in objects %}
+      {% if groupedItems[objectKey] is not defined %}
+          {% set groupedItems = groupedItems|merge({(objectKey): []}) %}
+          {% set objectTotals = objectTotals|merge({(objectKey): 0}) %}
+      {% endif %}
```

Comment:
Let's move this data manipulation to PHP. This shouldn't be done in Twig.

I'm having a hard time to reason about this Twig code, but this won't work. The objectTotals is just counting the loaded fields for each object. Is that right? Instead it must count how many fields are there in the database that may not be loaded in the array. 

Try to change the pagination from 30 to 5 items per page. You can see the pagination doesn't work.